### ISSUE TITLE
Instantiating GPUParticle2D nodes with a Line2D node on screen creates graphical artifacts

### ISSUE THREAD

**Issue by Lemonymous** [godotengine/godot#69669]
### Godot version

4.0.dev (30800d20f452015c054dc7d1d7cfd38362fe28ff)

### System information

Windows 10, Forward+, NVIDIA GeForce GTX 1050 Ti (31.0.15.1694)

### Issue description

Updating a Line2D node and creating/freeing GPUParticleEmitter2D nodes will cause graphical artifacts on Godot 4. The same test project built in Godot 3.5.1.stable does not create any artifacts.

### Steps to reproduce

- Create a Line2D with 2 points
- Make one of its points follow the mouse cursor
- Create a GPUParticle2D with a script freeing it when a Timer times out
- Instance GPUParticle2D nodes at the cursor location when holding down the mouse button
- Artifacts will appear when creating particles and moving the line around

### Minimal reproduction project

Project created in Godot 4 - this one will show artifacts
[GPU Particle Bug Godot 4.zip](https://github.com/godotengine/godot/files/10167584/GPU.Particle.Bug.Godot.4.zip)

The same project without created in Godot 3 - showing no artifacts
[GPU Particle Bug Godot 3.zip](https://github.com/godotengine/godot/files/10167583/GPU.Particle.Bug.Godot.3.zip)

**Comment by Calinou** [comment]
Related to https://github.com/godotengine/godot/issues/56691.

**Comment by Lemonymous** [comment]
On v4.0.beta.custom_build [291add339] after merged PR #70418,
the graphical artifacts produced in the minimal reproduction project in this Issue seems to no longer be an issue.

However, this code in the minimal reproduction project produces a whole wall of errors now, which it did not before. I have not investigated further.
```GDScript
extends Node2D

const EFFECT = preload("res://gpu_particles_2d.tscn")

func _input(event):
	if event.is_action_pressed("create_effect"):
		var effect = EFFECT.instantiate()
		add_child(effect)
		effect.position = get_global_mouse_position()
```

Errors:
![errors](https://user-images.githubusercontent.com/10490534/209043199-b9551a5a-10a7-4f7d-9075-05ae0d150127.png)

**Comment by akien-mga** [comment]
Fixed by #70418, errors are tracked in #70431.
